core/src/main/java/discord4j/core/event/dispatch/ChannelDispatchHandlers.java

```diff
@@ -45,6 +45,7 @@
         Channel.Type type = Channel.Type.of(context.getDispatch().getChannel().getType());
 
         switch (type) {
+            case GUILD_NEWS:
```

Comment:
This will result in a wrong type being returned. `GUILD_NEWS` and `GUILD_TEXT` are mostly the same, except `GUILD_NEWS` doesn't have slow mode properties.
A completely new type needs to be created and `TextChannel` and this new `NewsChannel` need to share another fundamental type for other methods (like `Guild#getSystemChannel`, `Webhook#getChannel`, etc.). We've proposed `GuildMessageChannel` as that's what makes the most intuitive sense.